lib/bookmark_query.rb

```diff
@@ -35,13 +35,22 @@ def list_all
     pms = Topic.private_messages_for_user(@user)
     results = results.merge(topics.or(pms))
 
-    results = results.merge(Post.secured(@guardian))
+    results = results.where(
+      "posts.post_type IN (?) OR bookmarks.post_id IS NULL", Topic.visible_post_types(@guardian&.user)
```

Comment:
Couldn't figure out a way to use `Post.secured` still and be able to do this so I just extracted it here